kv/kv.go

```diff
@@ -235,6 +235,11 @@ type Transaction interface {
 	SetDiskFullOpt(level kvrpcpb.DiskFullOpt)
 	// clear allowed flag
 	ClearDiskFullOpt()
+
+	// GetSavepoint gets the savepoint of the current transaction.
+	GetSavepoint() interface{}
```

Comment:
It's confusing to return `interface{}` here, please change it to a pre-defined structure/interface.

Reply:
I try to just return `*tikv.MemCheckpoint` here before, but there are many kind of implementation of this interface. Normally, each kind of implemention has its own `savepoint` struct, so I change this to `interface{}`.

Reply:
Better to avoid using `interface{}` in the interface definition, it is needed to wrap the transaction memory buffer status in the `KVTxn` layer maybe.
/cc @lcwangchao what do you think?